### PR TITLE
Changelog for 1.14.2

### PR DESCRIPTION
**Pull request by Benjin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Change Log
 
+## Version 1.14.2
+* Release date: May 20, 2022
+* Release status: GA
+
+### What's new in 1.14.2
+* Adds ability to create new SQL Docker instances when publishing a SQL database project
```
Comment (llali) [inline CHANGELOG.md:8]: why is this here? this feature is added to SQL project extension not mssql.

**Review by kburtram — APPROVED** [review]
Looks good to me.  For hotfixes I would typically add a line at the end like "Fix issues in milestone https://github.com/microsoft/vscode-mssql/milestone/24?closed=1".  Though that's optional.

File: CHANGELOG.md (hunk 1 above)
Comment (dzsquared) [inline CHANGELOG.md:8]: I'd suggest "Adds APIs for managing Azure resources", the note above would go in the SQL database project changelog/readme.

**Review by dzsquared — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Benjin) [reply CHANGELOG.md:8]: Ah, you're right.  I was thinking about the purpose of the release, not the actual contents of vscode-mssql.

**Comment by Benjin** [comment]
> Looks good to me. For hotfixes I would typically add a line at the end like "Fix issues in milestone https://github.com/microsoft/vscode-mssql/milestone/24?closed=1". Though that's optional.

@kburtram Since that milestone contains only the bug made just for approving the port, I think I'll leave it out this time.

**Review by dzsquared — APPROVED** [review]
